docs/model.md

```diff
@@ -0,0 +1,122 @@
+# Heapster Metric Model
+
+The Heapster Model is a structured representation of metrics for Kubernetes clusters, which is exposed through a set of REST API endpoints.
+It allows the extraction of up to 1 hour of historical data for any Container, Pod, Node or Namespace in the cluster, as well as the cluster itself.
+The model can also export the Average, Max and 95th Percentile for each one of these entities, over a duration of 1 minute, 1 hour or 24 hours.
+
+
+NOTICE: this document represents the state of the model at the next Heapster release, and is not representative of the current state of the master branch
+TODO(afein): remove NOTICE prior to next release
+
+## Usage
+
+The Heapster Model can be enabled by initializing Heapster with the `-use_model=true` flag. The resolution of the model can be configured through
+the `-model_resolution` flag, which will cause the model to store historical data at the specified resolution. If the `model_resolution` flag is not specified, the default
+resolution of 1 minute will be used.
+
+## API documentation
+
+A detailed documentation of each API endpoint is listed below. 
+
+All endpoints ending in `/metrics/{metric-name}` can accept the optional `start` and `end` query parameters 
+that represent the start and end time of the requested timeseries. The result
+will be a list of (Timestamp, Value) pairs in the time range (start, end].
+`start` and `end` are strings formatted according to RFC3339. If `start` is not
+defined, it is assumed as the zero Unix epoch time. If `end` is not defined,
+then all data later than `start` will be returned.
+
+### Cluster-level Metrics and Stats
+`/api/v1/model/`: The root path of the model API, displays all browsable subpaths.
+
+`/api/v1/model/metrics/`: Displays a list of available cluster-level metrics.
+
+`/api/v1/model/metrics/{metric-name}?start=X&end=Y`: Displays a set of (Timestamp, Value) 
+pairs for the requested cluster-level metric, between the time range specified by `start` and `end`. 
+
+`/api/v1/model/stats/`: Exposes the average, max and 95th percentile over the
+past minute, hour and day for each cluster metric.
+
+### Node-level Metrics and Stats
+`/api/v1/model/nodes`: Displays a list of all available nodes, along
```

Comment:
Can we define a top level resources supported and then just use that resources keyword everywhere else?